photon-lib/src/main/scala/com/linkedin/photon/ml/data/KeyValueScore.scala

```diff
@@ -76,49 +81,46 @@ protected[ml] class KeyValueScore(val scores: RDD[(Long, ScoredGameDatum)]) exte
   }
 
   /**
-   * Generic method to combine two [[KeyValueScore]] objects
+   * Generic method to combine two [[KeyValueScore]] objects.
    *
-   * @param op The operator to combine two scored data
-   * @param that [[KeyValueScore]] that needs to be merged with
-   * @return merged [[KeyValueScore]]
+   * @param op The operator to combine two [[ScoredGameDatum]]
+   * @param that The [[KeyValueScore]] to merge with this instance
+   * @return A merged [[KeyValueScore]]
    */
-  def binaryOp(
-    op: (ScoredGameDatum, ScoredGameDatum) => ScoredGameDatum,
-    that: KeyValueScore): KeyValueScore = {
-    val scores = this.scores
-      .cogroup(that.scores)
-      .flatMapValues {
-        case (Seq(sd1), Seq(sd2)) => Seq(op(sd1, sd2))
-        case (Seq(), Seq(sd2)) => Seq(op(sd2.getZeroScoreDatum, sd2))
-        case (Seq(sd1), Seq()) => Seq(op(sd1, sd1.getZeroScoreDatum))
-      }
-    new KeyValueScore(scores)
-  }
+  def binaryOp(op: (ScoredGameDatum, ScoredGameDatum) => ScoredGameDatum, that: KeyValueScore): KeyValueScore =
```

Comment:
More descriptive name? This is in effect implementing a join operation. Should the name reflect that?